feat(integrations): add support for google drive events

pashafateev opened the pull request:
- Adds support for watches being created for service accounts
- Does not support specifying `fileID` and `driveID` that is [here](https://linear.app/autokitteh/issue/ENG-1789/google-drive-events-add-support-for-fileid-and-driveid)
- Only supported events are `file_update` and `file_delete`. See ENG-1820

**refs:** [google drive event docs](https://github.com/autokitteh/docs.autokitteh.com/pull/131)

pashafateev committed:
add google drive event handler

pashafateev committed:
remove drivescope and clean up

pashafateev committed:
add support for service account watches

pashafateev committed:
remove unwanted file

itayd commented on integrations/google/creds.go:
```diff
@@ -81,12 +85,34 @@ func (h handler) handleCreds(w http.ResponseWriter, r *http.Request) {
+			googleoauth2.UserinfoEmailScope,
+			googleoauth2.UserinfoProfileScope,
+			googledrivev3.DriveFileScope,
+			// googledrivev3.DriveScope,
```
explain why commented out

tebeka committed:
feat(python): Add build export information (#809)

Fixes: ENG-1657

itayd committed:
feat(ENG-1303, ENG-1783)!: user objects (#825); kill envs

- refactor auth to use actual user objects with generated ids.
- when configured, users are generated when first logged in.
- authcontext always stores entire user, not just id.
- jwt tokens store only user id. these are verified as existing when
used.

currently users is an internal auth service. when we have proper rbac we
should consider making it external to enable orgs, etc.

This is based on #832 .

efiShtain committed:
fix: update webplatform to 2.54.0 (#841)

pashafateev committed:
fix file_deleted detection

efiShtain committed:
fix: isinstance called with multiple args (#843)

Co-authored-by: Daniel Abraham <daniel@autokitteh.com>
ENG-1827

daabr committed:
fix(tests): print in acceptance test project (#844)

Shouldn't clash with built-in function `print`

itayd committed:
fix(ENG-1773): allow to disable login redir (#847)

if requester specifies `?noredir=1`, instead of redirect on successful
login, it will just return 200.

itayd committed:
fix(1837): fix content-type (#848)

tebeka committed:
bug(python): User functions should not run as activities. (#839)

Fixes: ENG-1777

This changes the code to use a custom importer that will patch all files
in user code directory.

tebeka committed:
feat(python): Return custom values to ak (#842)

Fixes: ENG-1614

itayd committed:
kill env in db (#851)

efiShtain committed:
feat: remove legacy user id code after migration (#852)

we don't need this code anymore after the migration passed and all users
were migrated

itayd committed:
chore: update webplatform VERSION to 2.59.3 (#853)

itayd committed:
chore: remove unused temporalclient.LongRunning (trivial) (#856)

itayd committed:
fix: remove old web platform distributions on VERSION update (#855)

when running `make ak`, properly discard old downloaded versions.
currently it doesn't because bash is HARD, which causes the server to
possibly use the wrong version.

also:
- alert and disable webplatform when used version does not match
VERSION, do not use older versions.
- specifying the sha is no longer mandatory.
- upgrade to latest web-platform.

tebeka commented on integrations/google/drive/client.go:
```diff
@@ -37,3 +56,254 @@ func New(cvars sdkservices.Vars) sdkservices.Integration {
+		if err != nil {
+			return nil, err
+		}
+	}
```
```suggestion
	var src oauth2.TokenSource
	var err error
	if data.OAuthData != "" {
		src, err = oauthTokenSource(ctx, data.OAuthData)
	} else {
		src, err = jwtTokenSource(ctx, data.JSON)
	}
	if err != nil {
		return nil, err
	}
```

tebeka commented on integrations/google/drive/events.go:
```diff
@@ -0,0 +1,64 @@
+	eventType := "file_updated"
+	if latestEvent.Removed {
+		eventType = "file_deleted"
+	} // else if
```
Remove commented code.

pashafateev replied:
Done

mario99logic committed:
feat: Added gemini sdk  (#860)

This PR addresses ENG-1194. 

gemini_client simplifies the initialization of a GenerativeModel
instance from the google.generativeai SDK.
The function dynamically retrieves the API key based on the provided
connection name and configures the Generative AI client accordingly.

Key Changes: Added the gemini_client function

---------

Co-authored-by: Daniel Abraham <daniel@autokitteh.com>

mario99logic committed:
fix(python sdk): moved gemini client to google.py (#865)

Moved gemini_client to google.py and deleted gemini.py

---------

Co-authored-by: Daniel Abraham <daniel@autokitteh.com>

pashafateev committed:
docs(pythonrt):clarify token refresh logic (#863)

pashafateev committed:
Update integrations/google/drive/client.go

Co-authored-by: Miki Tebeka <miki@autokitteh.com>

pashafateev committed:
remove comment

efiShtain committed:
fix: switch ownerships to UUID userid (#869)

this change will set uuid user id in ownerships table so we can join on
it
it is backward compatible until we update the ownerships table itself 
once updated we should remove the extra check

tebeka committed:
chore(python): Add logging to gRPC handler (#862)

Fixes: ENG-1882

tebeka committed:
bug(python): Fail build on syntax errors (#859)

Fixes: ENG-1878

RonenMars committed:
chore: update web-platform version to: 2.83.0 (#872)

Previous version: 2.60.0.

List of changes, everything above:
https://github.com/autokitteh/web-platform/blob/main/CHANGELOG.md#2600-2024-11-20

daabr committed:
fix(github): OAuth code missing sometimes (#867)

When GitHub redirects OAuth flows back to us on success, sometimes the
`code` parameter is missing.

Instead of exchanging it for an OAuth token in order to get the app ID
(and other details), store the app ID as an environment variable, just
just the app name. We don't need OAuth tokens anyway, we generate JWTs.

On the way, we also report an accurate error message if the user tries
to initialize a GitHub connection with an app in a GitHub org, but isn't
an org admin.

Lastly, we also save more useful debug details in the connection vars.

Refs: INT-7

daabr committed:
cleanup(github): delete unused function wrappers for Starlark (#868)

Refs: INT-7 (stacked on top of #867)

dependabot[bot] committed:
chore(deps): bump golang.org/x/crypto from 0.27.0 to 0.31.0 (#877)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from
0.27.0 to 0.31.0.
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/golang/crypto/commit/b4f1988a35dee11ec3e05d6bf3e90b695fbd8909"><code>b4f1988</code></a>
ssh: make the public key cache a 1-entry FIFO cache</li>
<li><a
href="https://github.com/golang/crypto/commit/7042ebcbe097f305ba3a93f9a22b4befa4b83d29"><code>7042ebc</code></a>
openpgp/clearsign: just use rand.Reader in tests</li>
<li><a
href="https://github.com/golang/crypto/commit/3e90321ac7bcee3d924ed63ed3ad97be2079cb56"><code>3e90321</code></a>
go.mod: update golang.org/x dependencies</li>
<li><a
href="https://github.com/golang/crypto/commit/8c4e668694ccbaa1be4785da7e7a40f2ef93152b"><code>8c4e668</code></a>
x509roots/fallback: update bundle</li>
<li><a
href="https://github.com/golang/crypto/commit/6018723c74059e3b91c84268b212c2f6cdab1f64"><code>6018723</code></a>
go.mod: update golang.org/x dependencies</li>
<li><a
href="https://github.com/golang/crypto/commit/71ed71b4faf97caafd1863fed003e9ac311f10ee"><code>71ed71b</code></a>
README: don't recommend go get</li>
<li><a
href="https://github.com/golang/crypto/commit/750a45fe5e473d5afa193e9088f3d135e64eca26"><code>750a45f</code></a>
sha3: add MarshalBinary, AppendBinary, and UnmarshalBinary</li>
<li><a
href="https://github.com/golang/crypto/commit/36b172546bd03a74c79e109ec84c599b672ea9e4"><code>36b1725</code></a>
sha3: avoid trailing permutation</li>
<li><a
href="https://github.com/golang/crypto/commit/80ea76eb17c0c52f5d5d04e833d6aeb6b062d81d"><code>80ea76e</code></a>
sha3: fix padding for long cSHAKE parameters</li>
<li><a
href="https://github.com/golang/crypto/commit/c17aa50fbd32393e5d52fa65ca51cbfff0a75aea"><code>c17aa50</code></a>
sha3: avoid buffer copy</li>
<li>Additional commits viewable in <a
href="https://github.com/golang/crypto/compare/v0.27.0...v0.31.0">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=golang.org/x/crypto&package-manager=go_modules&previous-version=0.27.0&new-version=0.31.0)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)
You can disable automated security fix PRs for this repo from the
[Security Alerts
page](https://github.com/autokitteh/autokitteh/network/alerts).

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

daabr committed:
fix(google): env vars instead of hard-coded strings for pub/sub notifs (#879)

Example:

```
GMAIL_PUBSUB_TOPIC="projects/autokitteh-gapis-integration/topics/gmail-api-push"
GOOGLE_FORMS_PUBSUB_TOPIC="projects/autokitteh-gapis-integration/topics/forms-notifications"
```

Refs: INT-104,
https://github.com/autokitteh/docs.autokitteh.com/pull/136

pashafateev committed:
feat(integrations): add hubspot integration (#876)

Co-authored-by: Daniel Abraham <daniel@autokitteh.com>

daabr committed:
fix(google): verify signature of incoming push notifs (#880)

This is possible now because we've added service accounts in INT-104.

Refs: INT-111

mario99logic committed:
feat: add event handler for "app_home_opened" Slack (#873)

Added a new AppHomeOpenHandler to process app_home_opened events from
Slack. This handler validates the event payload and ignores
self-triggered events.

Refs: INT-93

---------

Co-authored-by: Miki Tebeka <miki@autokitteh.com>
Co-authored-by: Daniel Abraham <daniel@autokitteh.com>

tebeka committed:
feat(python): Add function argument names (#875)

Fixes ENG-1849

tebeka committed:
feat(python): Warn on I/O outside of an activity (#870)

Fixes: ENG-1893

tebeka committed:
bug(python): Stop run when there's a call error (#858)

Fixes: ENG-1696

daabr committed:
chore(python): update sdk (#886)

Update Python SDK and runner:
- New versions of `grpcio` and `google-generativeai` are no longer
incompatible
- update `PyGithub` to include our contribution:
https://github.com/PyGithub/PyGithub/pull/3082 (still pinned to commit,
not released yet)
- Remove obsolete Redis support
- Add HubSpot support instead

Refs: INT-99

efiShtain committed:
feat: add git to dockerfile to support git in python packages (#888)

tebeka committed:
fix(python): enable local imports (#889)

tebeka committed:
fix: maps.Keys is an iterator (#890)

efiShtain committed:
ci: build image on every pr (#891)

this makes docker builds an image on every pr
this is to solve the case where we merge to main
and only then we discover that the docker build step is broken

it does not publish the image, only builds

daabr committed:
docs: standardize license and notice files across public repos (#885)

Synced with https://github.com/autokitteh/kittehub/pull/130,
https://github.com/autokitteh/web-platform/pull/752.

Refs: INT-100

pashafateev committed:
remove second declaration of err

pashafateev committed:
Merge branch 'main' into pasha/eng-1557

pashafateev committed:
Merge branch 'main' into pasha/eng-1557

pashafateev commented:
replaced by #982
